File: 2163-kth-distinct-string-in-an-array/kth-distinct-string-in-an-array.java
import java.util.*;

class Solution {
    public String kthDistinct(String[] arr, int k) {
        Map<String, Integer> frequencyMap = new LinkedHashMap<>();
        
        // Count the frequency of each string
        for (String s : arr) {
            frequencyMap.put(s, frequencyMap.getOrDefault(s, 0) + 1);
        }
        
        // Find the kth distinct string
        int distinctCount = 0;
        for (Map.Entry<String, Integer> entry : frequencyMap.entrySet()) {
            if (entry.getValue() == 1) {
                distinctCount++;
                if (distinctCount == k) {
                    return entry.getKey();
                }
            }
        }
        
        // If there are fewer than k distinct strings
        return "";
    }
}

File: 2163-kth-distinct-string-in-an-array/kth-distinct-string-in-an-array.c
#include <string.h>
#include <stdlib.h>

char * kthDistinct(char ** arr, int arrSize, int k){
    int* frequency = (int*)calloc(arrSize, sizeof(int));
    
    // Count the frequency of each string
    for (int i = 0; i < arrSize; i++) {
        for (int j = 0; j < arrSize; j++) {
            if (strcmp(arr[i], arr[j]) == 0) {
                frequency[i]++;
            }
        }
    }
    
    // Find the kth distinct string
    int distinctCount = 0;
    for (int i = 0; i < arrSize; i++) {
        if (frequency[i] == 1) {
            distinctCount++;
            if (distinctCount == k) {
                free(frequency);
                return arr[i];
            }
        }
    }
    
    // If there are fewer than k distinct strings
    free(frequency);
    return "";
}